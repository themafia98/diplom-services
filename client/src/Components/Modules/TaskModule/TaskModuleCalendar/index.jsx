import React from "react";
import Scrollbars from 'react-custom-scrollbars';
import moment from "moment";
import TitleModule from "../../../TitleModule";
import { Calendar, Popover, Button } from "antd";

class TaskModuleCalendar extends React.PureComponent {
    getListData = value => {
        let listData = [];
        let today = moment().date();
        let valueDate = value.date();

        switch (valueDate) {
            case today: {
                listData = [{ type: "error", content: "Исправить баги. " }];
                break;
            }
            case today - 7: {
                listData = [
                    {
                        type: "success",
                        content: "Исправить баг. Исправить баги.Исправить баги. Исправить баги. Исправить баги.",
                    },
                ];
                break;
            }
            default:
        }
        return listData || [];
    };

    dateCellRender = value => {
        const listData = this.getListData(value);
        const content = listData.map(item => {
            return (
                <li onClick={this.test} key={item.content}>
                    {item.content}
                </li>
            );
        });

        const list = (
            <ul key="list" className="listTasks">
                {content}
            </ul>
        );
        let outValues = listData.length ? [...listData] : null;
        if (listData.length) {
            outValues = outValues.reduce((prev, current) => prev.content + current.content);
        }

        if (content.length)
            return (
                <Popover content={list}>
                    <Button>
                        <span className="calendarDate-content">{outValues.content}</span>
                    </Button>
                </Popover>
            );
    };

    test = event => { };

    render() {
        return (
            <Scrollbars>
                <div className="taskModuleCalendar">
                    <TitleModule classNameTitle="taskModuleTittle" title="Календарь задач" />
                    <div className="taskModuleCalendar__main">
                        <Calendar dateCellRender={this.dateCellRender} monthCellRender={this.monthCellRender} />
                    </div>
                </div>
            </Scrollbars>
        );
    }
}

export default TaskModuleCalendar;
