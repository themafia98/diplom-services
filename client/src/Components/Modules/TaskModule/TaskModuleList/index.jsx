import React from "react";
import Scrollbars from "react-custom-scrollbars";
import TableView from "../../../TableView";
import TitleModule from "../../../TitleModule";

class TaskModuleList extends React.PureComponent {
    render() {
        return (
            <Scrollbars>
                <div className="taskModule_all">
                    <TitleModule additional="Все задачи" classNameTitle="taskModuleTittle" title="Список всех задач" />
                    <div className="taskModuleAll_main">
                        <TableView path="searchTable" />
                    </div>
                </div>
            </Scrollbars>
        );
    }
}
export default TaskModuleList;
