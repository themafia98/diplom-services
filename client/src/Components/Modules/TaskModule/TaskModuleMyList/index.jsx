import React from "react";
import TableView from "../../../TableView";
import TitleModule from "../../../TitleModule";

class TaskModuleMyList extends React.PureComponent {
    render() {
        const { user, data = null, height, setCurrentTab } = this.props;

        return (
            <div ref={this.refModuleTask} className="taskModule_all">
                <TitleModule additional="Мои задачи" classNameTitle="taskModuleTittle" title="Список моих задач" />
                <div className="taskModuleAll_main">
                    <TableView
                        setCurrentTab={setCurrentTab}
                        height={height}
                        tasks={data ? data.tasks : null}
                        data={data}
                        flag={true}
                        user={user}
                        path="searchTable"
                    />
                </div>
            </div>
        );
    }
}
export default TaskModuleMyList;
