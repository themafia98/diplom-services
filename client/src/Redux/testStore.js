import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

/** stores for  jest */

export const initialState = {
  tabReducer: {
    parentSize: null,
    childrenSize: null,
    flag: false,
  },
  publicReducer: {
    status: 'online',
    prewStatus: 'online',
    requestError: null,
    caches: {},
  },
  router: {
    currentActionTab: 'mainModule',
    actionTabs: ['MainModule'],
    routeDataActive: {},
    routeData: {},
  },
};

const middlewares = [thunk.withExtraArgument({})]; /** test middlewares */
const mockStore = configureStore(middlewares); /** test config store */

const store = mockStore(initialState); /** init test store */

export default store;
