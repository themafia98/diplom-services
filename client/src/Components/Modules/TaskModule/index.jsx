import React from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import config from "../../../config.json";
import { connect } from "react-redux";
import Scrollbars from "react-custom-scrollbars";
import { Button, message } from "antd";

import { routeParser } from "../../../Utils";

import TabContainer from "../../TabContainer";
import { addTabAction, openPageWithDataAction } from "../../../Redux/actions/routerActions";
import { loadCurrentData } from "../../../Redux/actions/routerActions/middleware";
import TaskModuleCalendar from "./TaskModuleCalendar";
import TaskModuleList from "./TaskModuleList";
import TaskModuleMyList from "./TaskModuleMyList";
import TaskView from "./TaskView";
import CreateTask from "./CreateTask";

class TaskModule extends React.PureComponent {
    state = {
        height: null,
        heightController: null
    };

    static propTypes = {
        onErrorRequstAction: PropTypes.func.isRequired,
        path: PropTypes.string.isRequired,
        addTab: PropTypes.func.isRequired,
        onOpenPageWithData: PropTypes.func.isRequired,
        onLoadCurrentData: PropTypes.func.isRequired,
        publicReducer: PropTypes.object.isRequired,
        router: PropTypes.object.isRequired
    };

    componentDidMount = () => {
        const { onLoadCurrentData, path, visible } = this.props;
        const { height } = this.state;

        if (_.isNull(height) && !_.isNull(this.moduleTask) && visible) {
            const heightControllerForState = this.controller ? this.controller.getBoundingClientRect().height : null;
            const heightForState = this.moduleTask.getBoundingClientRect().height;
            this.setState({ ...this.state, height: heightForState, heightController: heightControllerForState });
        }
        if (visible) onLoadCurrentData({ path, storeLoad: "tasks", useStore: true, methodRequst: "GET" });
    };

    componentDidUpdate = prevProps => {
        const {
            visible,
            onLoadCurrentData,
            path,
            router: { shouldUpdate = false }
        } = this.props;
        const { height, heightController } = this.state;

        if (!_.isNull(height) && !_.isNull(this.moduleTask) && visible) {
            const heightControllerForState = this.controller ? this.controller.getBoundingClientRect().height : null;
            const heightForState = this.moduleTask.getBoundingClientRect().height;

            if (height !== heightForState || heightControllerForState !== heightController)
                this.setState({ ...this.state, height: heightForState, heightController: heightControllerForState });
        }

        if (shouldUpdate && visible)
            onLoadCurrentData({ path, storeLoad: "tasks", useStore: true, methodRequst: "GET" });
    };

    moduleTask = null;
    controller = null;
    refModuleTask = node => (this.moduleTask = node);
    refControllers = node => (this.controller = node);

    handlerNewTask = event => {
        const {
            addTab,
            setCurrentTab,
            router: { currentActionTab, actionTabs }
        } = this.props;

        if (currentActionTab !== "taskModule_createTask") {
            if (config.tabsLimit <= actionTabs.length)
                return message.error(`Максимальное количество вкладок: ${config.tabsLimit}`);
            const path = "taskModule_createTask";
            const isFind = actionTabs.findIndex(tab => tab === path) !== -1;

            if (!isFind) addTab(routeParser({ path }));
            else if (currentActionTab !== path) setCurrentTab(path);
        }
    };

    checkBackground = (path, mode = "default") => {
        const { actionTabs = [] } = this.props;
        if (mode === "default") return actionTabs.some(actionTab => actionTab === path);
    };

    getTaskByPath = path => {
        if (path) {
            const { height, heightController } = this.state;
            const isList = path === "taskModule_myTasks" || path === "taskModule_all";
            const {
                router,
                publicReducer: { status = null } = {},
                onOpenPageWithData,
                onLoadCurrentData,
                rest,
                setCurrentTab
            } = this.props;

            const isViewTask = path.startsWith("taskModule") && /__/gi.test(path);

            const isBackgroundTaskModuleAll = this.checkBackground("taskModule_all");
            const isBackgroundTaskModuleMyTasks = this.checkBackground("taskModule_myTasks");
            const isBackgroundTaskModuleCalendar = this.checkBackground("taskModule_сalendar");
            const isBackgroundTaskModuleCreateTask = this.checkBackground("taskModule_createTask");

            const route = routeParser({ pageType: "moduleItem", path });
            const isBackgroundTaskViewModule = _.isObject(route) && !_.isNull(route);

            const key = path.split("__")[1];

            return (
                <Scrollbars>
                    {isList ? (
                        <div key="controllers" ref={this.refControllers} className="controllersWrapper">
                            <Button className="newTaskButton" onClick={this.handlerNewTask} type="primary">
                                Создать новую задачу
                            </Button>
                        </div>
                    ) : null}
                    <TabContainer
                        className="tabList"
                        isBackground={isBackgroundTaskModuleAll}
                        visible={path === "taskModule_all"}
                    >
                        <TaskModuleList
                            key="taskList"
                            isBackground={isBackgroundTaskModuleAll}
                            rest={rest}
                            visible={path === "taskModule_all"}
                            setCurrentTab={setCurrentTab}
                            height={heightController ? height - heightController : height}
                            data={router.routeData[path]}
                        />
                    </TabContainer>
                    <TabContainer
                        className="tabList"
                        isBackground={isBackgroundTaskModuleMyTasks}
                        visible={path === "taskModule_myTasks"}
                    >
                        <TaskModuleMyList
                            key="myListTask"
                            rest={rest}
                            isBackground={isBackgroundTaskModuleMyTasks}
                            visible={path === "taskModule_myTasks"}
                            setCurrentTab={setCurrentTab}
                            height={height}
                            data={router.routeData[path]}
                            user="Павел Петрович"
                        />
                    </TabContainer>
                    <TabContainer
                        className="validateStyleWrapper"
                        isBackground={isBackgroundTaskModuleCalendar}
                        visible={path === "taskModule_сalendar"}
                    >
                        <TaskModuleCalendar
                            key="calendarTaskModule"
                            rest={rest}
                            isBackground={isBackgroundTaskModuleCalendar}
                            visible={path === "taskModule_сalendar"}
                            setCurrentTab={setCurrentTab}
                            onOpenPageWithData={onOpenPageWithData}
                            height={heightController ? height - heightController : height}
                            router={router}
                        />
                    </TabContainer>
                    <TabContainer
                        className="validateStyleWrapper"
                        isBackground={isBackgroundTaskModuleCreateTask}
                        visible={path === "taskModule_createTask"}
                    >
                        <CreateTask
                            key="createTask_module"
                            rest={rest}
                            isBackground={isBackgroundTaskModuleCreateTask}
                            visible={path === "taskModule_createTask"}
                            height={heightController ? height - heightController : height}
                            onLoadCurrentData={onLoadCurrentData}
                            statusApp={status}
                        />
                    </TabContainer>
                    <TabContainer
                        className="validateStyleWrapper"
                        isBackground={isBackgroundTaskViewModule}
                        visible={isViewTask}
                    >
                        <TaskView
                            key={key}
                            rest={rest}
                            isBackground={isBackgroundTaskViewModule}
                            visible={isViewTask}
                            height={heightController ? height - heightController : height}
                            onLoadCurrentData={onLoadCurrentData}
                            data={router.routeData[key]}
                        />
                    </TabContainer>
                </Scrollbars>
            );
        } else return <div>Not found path module</div>;
    };

    render() {
        const { path } = this.props;

        const component = this.getTaskByPath(path);
        return (
            <div key="taskModule" ref={this.refModuleTask} className="taskModule">
                {component ? component : null}
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        publicReducer: state.publicReducer,
        router: state.router
    };
};

const mapDispatchToProps = dispatch => {
    return {
        addTab: tab => dispatch(addTabAction(tab)),
        onOpenPageWithData: data => dispatch(openPageWithDataAction(data)),
        onLoadCurrentData: props => dispatch(loadCurrentData({ ...props }))
    };
};
export { TaskModule };
export default connect(mapStateToProps, mapDispatchToProps)(TaskModule);
