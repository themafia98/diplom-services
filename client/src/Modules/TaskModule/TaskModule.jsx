import React, { Component } from 'react';
import _ from 'lodash';
import { compose } from 'redux';
import { taskModuleType } from './TaskModule.types';
import { connect } from 'react-redux';
import { Button, message } from 'antd';
import { routeParser, oneOfType } from 'Utils';
import { settingsStatusSelector } from 'Redux/selectors';
import {
  addTabAction,
  openPageWithDataAction,
  removeTabAction,
  setActiveTabAction,
} from 'Redux/actions/routerActions';
import { loadCurrentData } from 'Redux/actions/routerActions/middleware';
import { loadCacheData } from 'Redux/actions/publicActions/middleware';
import entityRender from 'Utils/Tools/entityRender';
import withRouter from 'Components/Helpers/withRouter';
import types from 'types.modules';
import actionsTypes from 'actions.types';
import { moduleContextToProps } from 'Components/Helpers/moduleState';
import regExpRegister from 'Utils/Tools/regexpStorage';
import { withClientDb } from 'Models/ClientSideDatabase';
import actionPath from 'actions.path';
import { requestTemplate } from 'Utils/Api/api.utils';
import { ACTIONS } from 'App.constant';

class TaskModule extends Component {
  state = {
    height: null,
    heightController: null,
    path: null,
    counter: null,
    createTaskAvailable: false,
  };

  static propTypes = taskModuleType;

  static getDerivedStateFromProps = (props, state) => {
    if (props.path && props.path !== state.path && props.path.toLowerCase().includes('task')) {
      return {
        ...state,
        path: props.path,
      };
    }

    return state;
  };

  shouldComponentUpdate = (prevProps) => {
    const { path = '' } = this.state;
    const { path: pathProps } = this.props;

    return path === pathProps && !_.isEqual(this.props, prevProps);
  };

  componentDidMount = () => {
    const { path = '', appConfig, moduleContext } = this.props;
    const { visibility = false } = moduleContext;
    const { task: { limitList = 20 } = {} } = appConfig;
    const { height } = this.state;

    const isTaskModule = path && path.includes('task') && !path.split(regExpRegister.MODULE_ID)[1];
    if (height === null && this.moduleTask !== null && visibility) {
      this.recalcHeight();
    }

    if (isTaskModule) {
      const saveData = {
        current: 1,
        pageSize: limitList,
      };
      this.fetchTaskModule(null, {
        ...saveData,
        paginationState: saveData,
      });
    }

    this.setState({
      ...this.state,
      path,
    });

    window.addEventListener('resize', this.recalcHeight, false);
  };

  componentWillUnmount = () => {
    window.removeEventListener('resize', this.recalcHeight, false);
  };

  componentDidUpdate = () => {
    const { router, path = '', moduleContext, appConfig } = this.props;
    const { shouldUpdate = false, routeData = {} } = router;

    const { visibility = false } = moduleContext;
    const { isListCounterLoading = false, height } = this.state;
    const { task: { limitList = 20 } = {} } = appConfig;

    if (!visibility) return;

    const shouldBeInit = height === null && this.moduleTask !== null && visibility;
    if (shouldBeInit || [this.moduleTask, this.controller].every((type) => type !== null)) {
      this.recalcHeight();
    }

    const shouldUpdateList = routeData[path] && routeData[path]?.shouldUpdate;
    const isUnloadModule = shouldUpdate && !routeData[path]?.load;
    const { loading = false } = routeData[path] || {};

    const isCloseTabAction = !isUnloadModule && !shouldUpdateList && shouldUpdate && !loading;

    if (!isListCounterLoading && (isUnloadModule || shouldUpdateList || isCloseTabAction)) {
      const { saveData: saveDataState = null } = routeData[path] || {};
      const saveData = saveDataState
        ? saveDataState
        : {
            current: 1,
            pageSize: limitList,
          };

      this.shouldUpdateTasks() && this.fetchTaskModule(null, saveData, saveDataState);
    }
  };

  shouldUpdateTasks = () => {
    const { path = '' } = this.props;
    const viewId = path.split(regExpRegister.MODULE_ID)[1];
    return !!path && path.includes('task') && !viewId && !path.includes('createTask');
  };

  fetchTaskModule = async (customOptions = null, saveData = {}, saveDataState) => {
    const {
      onLoadCurrentData,
      path,
      udata: { _id: uid = '' } = {},
      modelsContext,
      clientDB,
      appConfig,
    } = this.props;
    const { counter = null, isListCounterLoading } = this.state;
    const { Request } = modelsContext;
    const { task: { limitList = 20 } = {} } = appConfig || {};

    const options = customOptions
      ? customOptions
      : {
          limitList,
          saveData,
        };

    if (!Request || isListCounterLoading) return;

    this.setState(
      {
        ...this.state,
        isListCounterLoading: true,
      },
      async () => {
        try {
          const { filteredInfo = null } = saveDataState || {};
          const arrayKeys = !filteredInfo
            ? []
            : Object.keys(filteredInfo).filter(
                (key) => filteredInfo[key]?.length && (key === 'editor' || key === 'date'),
              );
          if (!Request) return;
          const rest = new Request();
          const res = await rest.sendRequest(
            '/tasks/listCounter',
            'POST',
            {
              ...requestTemplate,
              actionType: actionsTypes.$CURRENT_LIST_COUNTER,
              moduleName: 'taskModule',
              params: {
                path,
                filterCounter: path.includes('all') ? null : uid,
                saveData: { ...saveDataState, arrayKeys },
              },
            },
            true,
          );
          if (res.status !== 200) throw new Error('Bad list');
          const { data: { response: { metadata = 0 } = {}, actions = [] } = {} } = res || {};

          const createTaskAvailable = actions.some((it) => it === ACTIONS.CREATE);

          if (counter !== metadata)
            this.setState({
              ...this.state,
              counter: metadata,
              createTaskAvailable,
            });
        } catch (error) {
          const { response = {} } = error || {};
          const { actions = [] } = response?.data || {};
          const createTaskAvailable = actions.some((it) => it === ACTIONS.CREATE);

          this.setState({ counter: 1, createTaskAvailable });
        }

        await onLoadCurrentData({
          action: actionPath.$LOAD_TASKS_LIST,
          path,
          options: {
            ...options,
            filterCounter: path.includes('taskModule_all') ? null : uid,
          },
          clientDB,
        });

        this.setState({
          ...this.state,
          isListCounterLoading: false,
        });
      },
    );
  };

  recalcHeight = () => {
    const { height, heightController, path = '' } = this.state;
    const { router: { currentActionTab = '' } = {} } = this.props;

    const isExists = this.controller?.current || !this.moduleTask?.current;

    if (currentActionTab !== path || !isExists || !this.moduleTask) {
      return;
    }

    const heightControllerForState = this.controller ? this.controller.getBoundingClientRect().height : null;
    const heightForState = this.moduleTask.getBoundingClientRect().height;

    if (height !== heightForState || heightControllerForState !== heightController)
      this.setState({
        ...this.state,
        height: window.innerHeight - 10 > heightForState ? heightForState - 250 : window.innerHeight - 250,
        heightController: heightControllerForState,
      });
  };

  moduleTask = null;
  controller = null;
  refModuleTask = (node) => (this.moduleTask = node);
  refControllers = (node) => (this.controller = node);

  handlerNewTask = (event) => {
    const {
      addTab,
      setCurrentTab,
      router: { currentActionTab, activeTabs },
      appConfig,
    } = this.props;
    const { tabsLimit = 50 } = appConfig;

    if (currentActionTab !== 'taskModule_createTask') {
      if (tabsLimit <= activeTabs.length)
        return message.error(`Максимальное количество вкладок: ${tabsLimit}`);
      const path = 'taskModule_createTask';
      const isFind = activeTabs.findIndex((tab) => tab === path) !== -1;
      const config = { hardCodeUpdate: false };
      if (!isFind) addTab(routeParser({ path }), config);
      else if (currentActionTab !== path) setCurrentTab(path, config);
    }
  };

  checkBackground = (path, visible, mode = 'default') => {
    const { activeTabs = [] } = this.props;
    if (mode === 'default') return !visible && activeTabs.some((actionTab) => actionTab === path);
  };

  renderTasksSubTabs = (subTabProps) => {
    const { path = '' } = this.state;
    const { entitysList = [], type, activeTabs = [] } = this.props;
    const { router: { routeData = {} } = {} } = this.props;

    const config = {
      validation: this.checkBackground,
      path,
      viewModuleName: 'taskViewModule',
      moduleName: 'taskModule',
      parentType: type,
      type: oneOfType(types.$sub_entrypoint_module, types.$entity_entrypoint),
    };

    const entityList = entityRender(
      activeTabs.filter((tab) => tab === entitysList),
      routeData,
      subTabProps,
      config,
    );

    return entityList.map(({ component = null }) => component);
  };

  getTaskByPath = (path) => {
    if (!path) return <div>Not found path module</div>;

    const { height, heightController, counter, createTaskAvailable } = this.state;
    const {
      router,
      router: { currentActionTab = '' } = {},
      status,
      onOpenPageWithData,
      onLoadCurrentData,
      onLoadCacheData,
      rest,
      removeTab,
      setCurrentTab,
      udata = {},
      statusList = {},
      onSetStatus = null,
      type: typeDefault = Symbol(''),
    } = this.props;

    const route = routeParser({ pageType: 'moduleItem', path });
    const [namePath = '', uuid = ''] = path.split(regExpRegister.MODULE_ID);
    const type = uuid ? types.$entity_entrypoint : typeDefault;

    const subTabProps = {
      currentActionTab,
      rest,
      statusApp: status,
      counter,
      setCurrentTab,
      loading: router?.routeData[path] && router?.routeData[path]?.loading,
      height: heightController ? height - heightController : height,
      data: router?.routeData[path],
      uuid,
      namePath,
      route,
      statusList,
      onSetStatus,
      onLoadCurrentData,
      onOpenPageWithData,
      onLoadCacheData,
      removeTab,
      router,
      udata,
    };

    return (
      <>
        {oneOfType(types.$sub_entrypoint_module, types.$entrypoint_module)(type) &&
        (path?.includes('all') || path?.includes('myTasks')) ? (
          <div key="controllers" ref={this.refControllers} className="controllersWrapper">
            <Button
              disabled={!createTaskAvailable}
              className="newTaskButton"
              onClick={this.handlerNewTask}
              type="primary"
            >
              Создать новую задачу
            </Button>
          </div>
        ) : null}
        {this.renderTasksSubTabs(subTabProps)}
      </>
    );
  };

  render() {
    const { path = '', type = '' } = this.props;

    const component = this.getTaskByPath(path);
    return (
      <div key="taskModule" ref={this.refModuleTask} className="taskModule">
        {component ? (
          component
        ) : (
          <div>{`Error render module, invalid module type is ${Symbol.keyFor(type)}`}</div>
        )}
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  const { router, publicReducer } = state;
  const { udata = {}, appConfig, status } = publicReducer;

  return {
    status,
    router,
    statusList: settingsStatusSelector(state, props),
    udata,
    appConfig,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    addTab: (tab, config = {}) => dispatch(addTabAction({ tab, config })),
    setCurrentTab: (tab, config = {}) => dispatch(setActiveTabAction({ tab, config })),
    removeTab: (tab) => dispatch(removeTabAction(tab)),
    onOpenPageWithData: (data) => dispatch(openPageWithDataAction(data)),
    onLoadCacheData: (props) => dispatch(loadCacheData(props)),
    onLoadCurrentData: (props) => dispatch(loadCurrentData(props)),
  };
};

export { TaskModule };
export default compose(
  withRouter,
  moduleContextToProps,
  withClientDb,
  connect(mapStateToProps, mapDispatchToProps),
)(TaskModule);
