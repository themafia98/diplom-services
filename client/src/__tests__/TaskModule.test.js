import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { initialState } from '../Redux/testStore';
import { TaskModule } from '../Components/Modules/TaskModule/TaskModule';

describe('<TaskModule /> template', () => {
  test('Should be render', () => {
    const props = {
      onErrorRequestAction: () => {},
      setCurrentTab: () => {},
      path: 'taskModule_all',
      addTab: () => {},
      onOpenPageWithData: () => {},
      onLoadCurrentData: () => {},
      ...initialState,
      router: { ...initialState.router, routeData: { taskModule__all: {} } },
    };
    const TaskModuleWrapper = shallow(<TaskModule {...props} />);
    expect(toJson(TaskModuleWrapper)).toMatchSnapshot();

    expect(TaskModuleWrapper.find('.newTaskButton').simulate('click', [])).toBeTruthy();

    const nextProps = {
      ...props,
      path: '',
    };

    const TaskModuleWrapperNoPath = shallow(<TaskModule {...nextProps} />);
    expect(toJson(TaskModuleWrapperNoPath)).toMatchSnapshot();
  });
});
