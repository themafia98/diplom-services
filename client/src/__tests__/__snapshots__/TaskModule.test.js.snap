// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskModule /> template Should be render 1`] = `
<div
  className="taskModule"
  key="taskModule"
>
  <Scrollbars
    autoHeight={false}
    autoHeightMax={200}
    autoHeightMin={0}
    autoHide={false}
    autoHideDuration={200}
    autoHideTimeout={1000}
    hideTracksWhenNotNeeded
    renderThumbHorizontal={[Function]}
    renderThumbVertical={[Function]}
    renderTrackHorizontal={[Function]}
    renderTrackVertical={[Function]}
    renderView={[Function]}
    tagName="div"
    thumbMinSize={30}
    universal={false}
  >
    <div
      className="controllersWrapper"
      key="controllers"
    >
      <Button
        block={false}
        className="newTaskButton"
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        Создать новую задачу
      </Button>
    </div>
    <TabContainer
      className="tabList"
      isBackground={false}
      isPortal={false}
      isTab={false}
      key="mainModuletaskModule_alltaskList_tab"
      onChangeVisibleAction={null}
      visible={true}
    >
      <TaskModuleList
        counter={null}
        data={Object {}}
        height={null}
        isBackground={false}
        key="taskModule_listAll"
        loaderMethods={Object {}}
        rest={Object {}}
        router={
          Object {
            "activeTabs": Array [
              "MainModule",
            ],
            "currentActionTab": "mainModule",
            "routeData": Object {
              "taskModule__all": Object {},
            },
            "routeDataActive": Object {},
          }
        }
        setCurrentTab={[Function]}
        statusApp="online"
        urrentActionTab="mainModule"
        visible={true}
      />
    </TabContainer>
    <TabContainer
      className="tabList"
      isBackground={false}
      isPortal={false}
      isTab={false}
      key="mainModuletaskModule_allmyTaskList_tab"
      onChangeVisibleAction={null}
      visible={false}
    >
      <TaskModuleMyList
        counter={null}
        currentActionTab="mainModule"
        data={Object {}}
        height={null}
        isBackground={false}
        key="myList_tasks"
        loaderMethods={Object {}}
        rest={Object {}}
        router={
          Object {
            "activeTabs": Array [
              "MainModule",
            ],
            "currentActionTab": "mainModule",
            "routeData": Object {
              "taskModule__all": Object {},
            },
            "routeDataActive": Object {},
          }
        }
        setCurrentTab={[Function]}
        statusApp="online"
        udata={Object {}}
        visible={false}
      />
    </TabContainer>
    <TabContainer
      className="validateStyleWrapper"
      isBackground={false}
      isPortal={false}
      isTab={false}
      key="calendarTaskModule_tab"
      onChangeVisibleAction={null}
      visible={false}
    >
      <TaskModuleCalendar
        data={Object {}}
        height={null}
        isBackground={false}
        key="calendarTaskModule"
        loaderMethods={Object {}}
        onLoadCurrentData={[Function]}
        onOpenPageWithData={[Function]}
        onSetStatus={null}
        rest={Object {}}
        router={
          Object {
            "activeTabs": Array [
              "MainModule",
            ],
            "currentActionTab": "mainModule",
            "routeData": Object {
              "taskModule__all": Object {},
            },
            "routeDataActive": Object {},
          }
        }
        setCurrentTab={[Function]}
        statusApp="online"
        statusList={Object {}}
        udata={Object {}}
        visible={false}
      />
    </TabContainer>
    <TabContainer
      className="validateStyleWrapper"
      isBackground={false}
      isPortal={false}
      isTab={false}
      key="createTask_module_tab"
      onChangeVisibleAction={null}
      visible={false}
    >
      <CreateTask
        isBackground={false}
        key="createTaskModule"
        onLoadCurrentData={[Function]}
        onOpenPageWithData={[Function]}
        onSetStatus={null}
        removeTab={[Function]}
        rest={Object {}}
        router={
          Object {
            "activeTabs": Array [
              "MainModule",
            ],
            "currentActionTab": "mainModule",
            "routeData": Object {
              "taskModule__all": Object {},
            },
            "routeDataActive": Object {},
          }
        }
        statusApp="online"
        statusList={Object {}}
        udata={Object {}}
        visible={false}
        visibleMode="default"
      />
    </TabContainer>
    <TabContainer
      className="validateStyleWrapper"
      isBackground={false}
      isPortal={false}
      isTab={false}
      key="taskView_tab"
      onChangeVisibleAction={null}
      visible={false}
    >
      <Connect(TaskView)
        height={null}
        isBackground={false}
        key="taskViewModule"
        onLoadCacheData={[Function]}
        onLoadCurrentData={[Function]}
        onOpenPageWithData={[Function]}
        rest={Object {}}
        setCurrentTab={[Function]}
        uuid=""
        visible={false}
      />
    </TabContainer>
  </Scrollbars>
</div>
`;

exports[`<TaskModule /> template Should be render 2`] = `
<div
  className="taskModule"
  key="taskModule"
>
  <div>
    Not found path module
  </div>
</div>
`;
